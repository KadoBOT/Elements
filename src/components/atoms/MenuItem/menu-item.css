.menu-item > a {
  cursor: pointer;
}

.menu-item:hover {
  color: #ec0f50 !important;
  border-bottom: 2px solid #ec0f50;
}
