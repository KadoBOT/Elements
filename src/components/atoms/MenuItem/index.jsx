import React, {PropTypes} from 'react'

import { Text } from '../../atoms/Text'
import './menu-item.css'

const MenuItem = ({item}) => {
  return(
    <Text className="menu-item">
      <a href={`#${item}`}>{item}</a>
    </Text>
  )
}

MenuItem.propTypes = {
  props: PropTypes.type
}

export default MenuItem
