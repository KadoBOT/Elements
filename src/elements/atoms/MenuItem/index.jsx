import React, {PropTypes} from 'react'

import { Text } from '../../atoms/Text'
import './menu-item.css'

const MenuItem = ({item}) => (
  <Text className="menu-item">
    <a href={`#${item}`}>{item}</a>
  </Text>
)

MenuItem.propTypes = {
  item: PropTypes.string
}

export default MenuItem
