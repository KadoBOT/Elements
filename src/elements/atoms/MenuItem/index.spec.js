import React from 'react'
import { shallow, render, mount } from 'enzyme';

import MenuItem from './'

it('should render the MenuItem', () => {
    const wrapper = shallow(
        <MenuItem item="blog" />
    );
    expect(wrapper).toMatchSnapshot();
});
