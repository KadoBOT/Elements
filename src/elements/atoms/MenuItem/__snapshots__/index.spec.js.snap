exports[`test should render the MenuItem 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.p
    className="menu-item">
    <a
        href="#blog">
        blog
    </a>
</styled.p>,
  "nodes": Array [
    <styled.p
      className="menu-item">
      <a
            href="#blog">
            blog
      </a>
</styled.p>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        item="blog" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": "blog",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.p
          className="menu-item">
          <a
                    href="#blog">
                    blog
          </a>
</styled.p>,
        "_debugID": 2,
        "_renderedOutput": <styled.p
          className="menu-item">
          <a
                    href="#blog">
                    blog
          </a>
</styled.p>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    item="blog" />,
}
`;
